docs/userguide/core-workflow.md

```diff
@@ -1,297 +1,78 @@
-      should be subclassed to implement Workflows that have [private state](#private-state).
-    - [**`StatelessWorkflow`**](/workflow/kotlin/api/gfmCollector/workflow/com.squareup.workflow1/-stateless-workflow/)
-      should be subclassed to implement Workflows that _don't_ have any private state. See [Stateless Workflows](#stateless-workflows).
+* State includes a `List<MessageId>`, and the selected `MessageId`
+* Rendering is a `SplitScreen` view model, to be assembled from the renderings of the other two Workflows
+* Accepts no Props, and emits no Output
 
-Workflows have several responsibilities:
+![Workflow schematic showing MessageWorkflow](../images/email_browser_workflow_schematic.svg)
 
-## Workflows have state
+When `EmailBrowserWorkflow` is asked to provide its Rendering, it in turn asks for Renderings from its two children.
 
-Once a Workflow has been started, it always operates in the context of some state. This state is
-divided into two parts: private state, which only the Workflow implementation itself knows about,
-which is defined by the `State` type, and properties (or "props"), which is passed to the Workflow
-from its parent (more on hierarchical workflows below).
+* It provides the `List<MessageId>` from its state as the Props for `EmailInboxWorkflow` and receives an `InBoxScreen` rendering in return. That `InboxScreen` becomes the left pane of a `SplitScreen` Rendering.
+* For the `SplitScreen`'s right pane, the browser Workflow provides the currently selected `MessageId` as input to `EmailMessageWorkflow`, to get a `MessageScreen` rendering.
 
-### Private state
+![Workflow schematic showing EmailBrowserWorkflow rendering by delegating to two children, InboxWorkflow and MessageWorkflow, and assembling their renderings into its own.](../images/split_screen_schematic.svg)
 
-Every Workflow implementation defines a `State` type to maintain any necessary state while the
-workflow is running.
+!!! note
+    Note that the two children, `EmailInboxWorkflow` and `EmailMessageWorkflow`, have no knowledge of each other, nor of the context in which they are run.
 
-For example, a tic-tac-toe game might have a state like this:
+The `InboxScreen` rendering includes an `onMessageSelected(MessageId)` function.
+When that is called, `EmailInboxWorkflow` emits the given `MessageId` as Output to be processed by its parent.
+`EmailBrowserWorkflow`  receives that output, and updates the `selection: MessageId` of its State accordingly.
 
-=== "Swift"
-    ```Swift
-    struct State {
+![Workflow schematic showing EmailBrowserWorkflow rendering by delegating to two children, InboxWorkflow and MessageWorkflow, and assembling their renderings into its own.](../images/split_screen_update.svg)
 
-        enum Player {
-            case x
-            case o
-        }
+Whenever such a [Workflow Action cascade](../../glossary#action-cascade) fires, the root Workflow is asked for a new Rendering.
```

Comment:
Since you use 'Action Cascade' explicitly here - I think we should specify actions above, otherwise the gap is too big.

Reply:
Yeah, I struggled with that. This is an improvement, thanks.